### PR TITLE
Feature/833 cob windowing

### PR DESCRIPTION
**Pull request by JulianHammerl**
* **Tickets addressed:** MAXGNC-833
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

The centerOfBrightness module was updated to accept the input parameters for a masking window, and the module then applies that window to only consider pixels within the window for the computation of the center of brightness.

## Verification

The unit test was updated to check the windowing approach.

## Documentation

The documentation was updated accordingly.

## Future work

At the moment, swig_eigen.i doesn't include the Eigen::Vector2i type, so Eigen::VectorXi was used instead for the corresponding windowing input parameters. This should be changed once the swig_eigen.i file has been updated.

### CODE REVIEW

**Review by thibaudteil — APPROVED** [review]
Good work!

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py
Comment (patkenneally) [inline src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py]: :large_orange_diamond: Remove all these comments.

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py
Comment (patkenneally) [inline src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py]: :large_orange_diamond: Above, `unitTestSupport` is an unused import.

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py
Comment (patkenneally) [inline src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py]: :large_orange_diamond: Remove the line between the decorator and the test function

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py
Comment (patkenneally) [inline src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py]: :large_orange_diamond: What do you think about us changing the name of the `valid` parameter to better indicate it's purpose? I found I needed to read down to line 120 to work out what it was used for.

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py
Comment (patkenneally) [inline src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py:104]: :large_orange_diamond: By the looks of it, this entire conditional for image "half_half.png" can move down into the conditional with the assert at the end of this file. There's no usage that requires it here.

**Review by patkenneally — APPROVED** [review]
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

Nice PR. I particularly appreciate your clean up by changing the implicit conversions to explicit.

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py
Comment (JulianHammerl) [reply src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py]: Removed it.

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py
Comment (JulianHammerl) [reply src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py]: Removed them.

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py
Comment (JulianHammerl) [reply src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py]: Renamed it to `validImage`.

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py
Comment (JulianHammerl) [reply src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py]: Done.

File: src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py
Comment (JulianHammerl) [reply src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py:104]: I agree with you for the scope of this PR. However, the truth values are computed in this conditional. In the dependent PR #256 , the truth values for brightness are computed during the sequential sim execution, so this information is needed earlier than in the assert at the end of the file. In my opinion I should leave it like it is for now then? As you said in PR #256 , this UnitTest might have to be changed in a major way anyways.

Comment (patkenneally) [reply src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py:104]: Agreed

**Pull request merged**
